modules/scripts/startup-script/main.tf

```diff
@@ -20,11 +20,16 @@ locals {
 }
 
 locals {
-  ops_agent_installer = var.install_cloud_ops_agent ? [{
-    type        = "shell"
-    source      = "${path.module}/files/install_cloud_ops_agent.sh"
-    destination = "install_cloud_ops_agent_automatic.sh"
-  }] : []
+  monitoring_agent_installer = (
+    var.install_cloud_ops_agent || var.install_stackdriver_agent ?
+    [{
+      type        = "shell"
+      source      = "${path.module}/files/install_monitoring_agent.sh"
+      destination = "install_monitoring_agent_automatic.sh"
+      args        = var.install_cloud_ops_agent ? "false" : "true" # install legacy (stackdriver)
```

Comment:
I think Terraform would render `!var.install_cloud_ops_agent` into a string, FYI, but I agree with explicitness here. +1.

That said, I think the script and this section would be better if instead of "true" you used "legacy".